Frameworks/Foundation/NSOperation.mm

```diff
@@ -84,11 +84,32 @@ - (void)_checkReady {
     }
 }
 
+- (void)_completionBlockThread {
+    decltype(_completionBlock) localCompletion;
+
+    { // _completionBlockLock scope
+        std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+        localCompletion = std::move(_completionBlock);
```

Comment:
_completionBlock in this case is undefined, which doesn't mean that it cannot go to a reliable state, just that code should not assume any specific state.  What it means in this specific case is that after the std::move(), you should set _completionBlock to nil even if you know that our implementation will set it to nil.  

---
In reply to: [73055468](https://github.com/Microsoft/WinObjC/pull/784#discussion_r73055468) [](ancestors = 73055468)